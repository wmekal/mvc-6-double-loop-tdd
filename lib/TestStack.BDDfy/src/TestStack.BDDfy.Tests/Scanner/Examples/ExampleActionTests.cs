using ApprovalTests;
using Shouldly;
using TestStack.BDDfy.Reporters;
using Xunit;

namespace TestStack.BDDfy.Tests.Scanner.Examples
{
    public class ExampleActionTests
    {
        private int _value;

        [Fact]
        public void CanUseActionsInExamples()
        {
            ExampleAction actionToPerform = null;
            int valueShouldBe = 0;
            var story = this.Given(_ => SomeSetup())
                .When(() => actionToPerform)
                .Then(_ => ShouldBe(valueShouldBe))
                .WithExamples(new ExampleTable("Action to perform", "Value should be")
                {
                    { new ExampleAction("Do something", () => { _value = 42; }), 42 },
                    { new ExampleAction("Do something else", () => { _value = 7; }), 7 }
                })
                .BDDfy();


            var textReporter = new TextReporter();
            textReporter.Process(story);
            Approvals.Verify(textReporter.ToString());
        }

        private void ShouldBe(int i)
        {
            _value.ShouldBe(i);
        }

        private void SomeSetup()
        {
            
        }
    }
}