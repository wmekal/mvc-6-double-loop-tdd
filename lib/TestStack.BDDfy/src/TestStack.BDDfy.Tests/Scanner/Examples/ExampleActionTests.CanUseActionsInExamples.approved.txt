
Scenario: Can use actions in examples
	Given some setup
	When  <Action to perform>
	Then should be <Value should be>

Examples: 
| Action to perform | Value should be |
| Do something      | 42              |
| Do something else | 7               |

